src/Admin/ProductBlocksService.php

```diff
@@ -67,13 +72,16 @@ class ProductBlocksService implements Service, Registerable, Conditional {
 	 * ProductBlocksService constructor.
 	 *
 	 * @param AssetsHandlerInterface $assets_handler
+	 * @param ProductHelper          $product_helper
 	 * @param AttributeManager       $attribute_manager
 	 * @param MerchantCenterService  $merchant_center
 	 */
-	public function __construct( AssetsHandlerInterface $assets_handler, AttributeManager $attribute_manager, MerchantCenterService $merchant_center ) {
+	public function __construct( AssetsHandlerInterface $assets_handler, ProductHelper $product_helper, AttributeManager $attribute_manager, MerchantCenterService $merchant_center ) {
 		$this->assets_handler    = $assets_handler;
 		$this->attribute_manager = $attribute_manager;
 		$this->merchant_center   = $merchant_center;
+
+		$this->channel_visibility_block = new ChannelVisibilityBlock( $product_helper );
```

Comment:
> What's the reason to initiate the Class `ChannelVisibilityBlock` here instead of Core ServiceProvider?

Since I'm still not quite sure what a "service" is in the `league/container` package and what kind of class should belong to "service". When I was adding the `ChannelVisibilityBlock` class, I thought it might not be a "service", so I used this approach.

> And then you pass the ChannelVisibilityBlock as a param here: [...]

Thanks for the suggestion. Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/2214/commits/177e5ab911aed0798d711055321cbd1da59ea1c6.

Reply:
> Since I'm still not quite sure what a "service" is in the league/container package and what kind of class should belong to "service". When I was adding the ChannelVisibilityBlock class, I thought it might not be a "service", so I used this approach.

Here there is a nice explanation: https://carlalexander.ca/dependency-injection-wordpress/  (section: On service classes and singletons). Another advantage of having the class `ChannelVisibilityBlock` as a dependency is that it's easier to test since you can simply mock the dependencies you pass to `ProductBlocksService`.